pygfx/utils/transform.py

```diff
@@ -524,6 +524,17 @@ class AffineTransform(AffineBase):
     is_camera_space : bool
         If True, the transform represents a camera space which means that it's
         ``forward`` and ``right`` directions are inverted.
+    state_basis : str
+        One of "components" or "matrix". If "components", the transform will
+        store its state in the form of position, rotation, and scale. If "matrix",
+        the transform will store its state in the form of a matrix. The default
+        is "components". This can have performance implications when the
+        transform is used in a context where the state is updated in one
+        form but queried in another.
```

Comment:
Maybe a hint for a use-case where this would be handy? For users ... and for me 😄

Reply:
In certain software applications animations are pregenerated as a sequence of transform matrices. It's much more efficient to be able to just set them and use them as is without any computational overhead at all.

Reply:
I've reviewed all docstrings and type annotations in transform.py. Could you take another look if it's clear now?